azurerm/internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -337,25 +333,28 @@ func resourcePostgresqlFlexibleServerRead(d *pluginsdk.ResourceData, meta interf
 	if props := resp.ServerProperties; props != nil {
 		d.Set("administrator_login", props.AdministratorLogin)
 		d.Set("zone", props.AvailabilityZone)
-		if props.SourceServerName != nil && props.SourceSubscriptionID != nil && props.SourceResourceGroupName != nil {
-			d.Set("source_server_id", parse.NewFlexibleServerID(*props.SourceSubscriptionID, *props.SourceResourceGroupName, *props.SourceServerName).ID())
-		}
 		d.Set("version", props.Version)
-		d.Set("cmk_enabled", props.ByokEnforcement)
 		d.Set("fqdn", props.FullyQualifiedDomainName)
-		d.Set("public_network_access_enabled", props.PublicNetworkAccess == postgresqlflexibleservers.ServerPublicNetworkAccessStateEnabled)
+		d.Set("public_network_access_enabled", props.Network.PublicNetworkAccess == postgresqlflexibleservers.ServerPublicNetworkAccessStateEnabled)
 
-		if props.DelegatedSubnetArguments != nil {
-			d.Set("delegated_subnet_id", props.DelegatedSubnetArguments.SubnetArmResourceID)
+		if props.SourceServerResourceID != nil {
+			d.Set("source_server_id", props.SourceServerResourceID)
+		}
+
+		if props.Network.DelegatedSubnetResourceID != nil {
+			d.Set("delegated_subnet_id", props.Network.DelegatedSubnetResourceID)
 		}
 
 		if err := d.Set("maintenance_window", flattenArmServerMaintenanceWindow(props.MaintenanceWindow)); err != nil {
 			return fmt.Errorf("setting `maintenance_window`: %+v", err)
 		}
 
-		if storage := props.StorageProfile; storage != nil {
-			d.Set("storage_mb", storage.StorageMB)
-			d.Set("backup_retention_days", storage.BackupRetentionDays)
+		if storage := props.Storage; storage != nil {
```

Comment:
storage!=nil && storage.storageSizeGB!=nil?

Reply:
Also checking if storage.storageSizeGB is nil now. Thanks